File: alogorithm/java_algorithm/src/_2020/_01/_30_HammingDistance.java
package _2020._01;

// https://leetcode.com/problems/hamming-distance/
public class _30_HammingDistance {
    public static void main(String[] args) {
        System.out.println(new _30_HammingDistance().hammingDistance(1, 4));
    }

    public int hammingDistance(final int x, final int y) {
        final int xor = x ^ y;
        return Integer.bitCount(xor);
    }
}

File: alogorithm/rust_algorithm/src/_2020/_01/_30_Hamming_Distance.rs
use std::ops::BitXor;

// https://leetcode.com/problems/hamming-distance/
impl Solution {
    pub fn hamming_distance(x: i32, y: i32) -> i32 {
        let xor = x.bitxor(y) as u32;
        return xor.count_ones() as i32;
    }
}
